### PR TITLE
Added Authentication Functionality

### PR DESCRIPTION
**Pull request by SRUJANA-PENUGONDA13**
Added Authentication Functionality

1. Added validations for signup and signin page
2. Added authentication for cart and wishlist page
3. Updated buttons to update quantity in cart card in cart page

https://user-images.githubusercontent.com/72279128/167298854-a351e960-f6d5-4c37-8f5c-25dec0e4e8c3.mp4

### CODE REVIEW

File: src/frontend/services/signupService.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+const signupService = async ({ name, email, password }) => {
+  try {
+    const {
+      data: { encodedToken },
+    } = await axios.post(`/api/auth/signup`, {
+      firstName: name,
+      lastName: "",
```
Comment (LordUttam) [inline src/frontend/services/signupService.js]: Instead of doing this, why not just remove this field from the model?

**Review by LordUttam — CHANGES_REQUESTED** [review]
On refreshing we are being logged out which should not happen. Read the necessary values from local storage in context.

I noticed a small bug that when I logout and login again, my cart and wishlist are shown as empty until I add something to them.

File: src/frontend/services/signupService.js (hunk 1 above)
Comment (SRUJANA-PENUGONDA13) [reply src/frontend/services/signupService.js]: Updated

**Comment by SRUJANA-PENUGONDA13** [comment]
@LordUttam Thank You for reviewing

**Review by irajbhinde — COMMENTED** [review]
After login user is redirected to home page, but at home page there is no logout,
 if we go to explore & then come back to home, then the logout is visible, check why is that happening, user should be able to see logout after signing in. Otherwise, code looks clean & looks good to merge

**Pull request merged**